syntax = "proto3";

package LibreASR;

service LibreASR {

  rpc Transcribe(AudioAndSettings) returns (Transcript) {}
  rpc TranscribeStream(stream AudioOrSettings) returns (stream Event) {}
  rpc PreloadVoice(AudioAndSettings) returns (VoiceClip) {}

  rpc Translate(Text) returns (Text) {}
}

message Audio {
  bytes data = 1;
}

message Settings {
  int32 sr = 1;
  string lang = 2;
  string preload_voice_id = 3;
}

message AudioAndSettings {
  Audio audio = 1;
  Settings settings = 2;
}

message AudioOrSettings {
  oneof content {
    Audio audio = 1;
    Settings settings = 2;
  }
}

message Transcript {
  string data = 1;
}

message TranscriptEvent {
  string transcript = 1;
}

message SentenceEvent {
  string transcript = 1;
}

message Event {
  oneof content {
    TranscriptEvent te = 1;
    SentenceEvent se = 2;
  }
}

message VoiceClip {
  string id = 1;
}

message Text {
  string src = 1;
  string tgt = 2;
  string text = 3;
  string model = 4;
}